### PR TITLE
implement aligned realloc, use memalign, allow use on embedded

### PR DESCRIPTION
**Pull request by corecode**
Quick 2 commits.  Suggest fast forward instead of merge.

### CODE REVIEW

**Comment by daniel5151** [comment]
Sure, these look good to me.

Any more coming, or do you want me to publish these as a 1.0.5?

File: Cargo.toml [hunk 1]
```diff
@@ -12,4 +12,7 @@ repository  = "https://github.com/daniel5151/libc_alloc"
 keywords = ["malloc", "memalign", "libc", "allocator", "no_std"]
 categories = ["development-tools::ffi", "embedded", "no-std"]
 
+[features]
+libc = []
```
Comment (daniel5151) [inline Cargo.toml]: mind adding a comment explaining what this feature does?

**Review by daniel5151 — CHANGES_REQUESTED** [review]
Ah, though, some documentation would be good...

**Comment by corecode** [comment]
Yes, I'll add some docs.

Maybe we should default to "libc" if windows isn't set?  After all, the crate name is "libc_alloc".

File: src/lib.rs [hunk 2]
```diff
@@ -16,10 +16,11 @@
 #![no_std]
 
 use core::alloc::{GlobalAlloc, Layout};
+use core::cmp;
 use core::ffi::c_void;
 use core::ptr;
 
-#[cfg(target_family = "unix")]
+#[cfg(any(target_family = "unix", feature = "libc"))]
```
Comment (daniel5151) [inline src/lib.rs]: Now that I think about it... maybe this should just be

```rust
#[cfg(not(target_family = "windows"))]
```

and avoid introducing the new feature entirely?

**Review by daniel5151 — CHANGES_REQUESTED** [review]
Left a suggestion that might solve the issue you identified without requiring a new feature + additional docs.

Let me know what you think

File: src/lib.rs (hunk 2 above)
Comment (corecode) [reply src/lib.rs]: I prefer your suggestion.

**Comment by corecode** [comment]
What do you think about this?

**Comment by daniel5151** [comment]
Is there a reason you're switching from `posix_memalign` to just `memalign`?

If anything, we should be duplicating this logic from the stdlib?

https://doc.rust-lang.org/src/std/sys/unix/alloc.rs.html#55-102

**Comment by corecode** [comment]
Like I wrote in the commit, newlib does not provide posix_memalign.

While glibc's manpage states:
 > POSIX requires that memory obtained from posix_memalign() can be freed using free(3).  Some systems provide no way to reclaim memory allocated  with  memalign() or valloc() (because one can pass to free(3) only a pointer obtained from malloc(3), while, for example, memalign() would call malloc(3) and then align the obtained value).  The glibc implementation allows memory obtained from any of these functions to be reclaimed with free(3).

I think that those systems are very rare, and they won't have posix_memalign either.

**Comment by daniel5151** [comment]
Ahh, didn't catch that extra context in the commit messages, my bad 😅

Sure, in that case, this all looks good to me.
I'll go ahead and land these commits.

Thanks for the PR!

* * *

Like I asked above: Any more changes coming, or would you like me to to publish a new 1.0.5 release?

**Comment by corecode** [comment]
No changes from me at the moment, good to release.

Thanks!

**Comment by daniel5151** [comment]
Cool, I've published 1.0.5 to crates.io.

Cheers!

**Pull request merged**